tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/SendMessageBatchAsyncIntegration.cs

```diff
@@ -52,19 +52,26 @@ public class SendMessageBatchAsyncIntegration
             // for the InjectHeadersIntoMessage<TSendMessageRequest> call below
             var requestProxy = request.DuckCast<ISendMessageBatchRequest>();
 
+            var queueName = AwsSqsCommon.GetQueueName(requestProxy.QueueUrl);
             var scope = AwsSqsCommon.CreateScope(Tracer.Instance, Operation, out var tags, spanKind: SpanKinds.Producer);
             if (tags is not null && requestProxy.QueueUrl is not null)
             {
                 tags.QueueUrl = requestProxy.QueueUrl;
-                tags.QueueName = AwsSqsCommon.GetQueueName(requestProxy.QueueUrl);
+                tags.QueueName = queueName;
             }
 
-            if (scope?.Span?.Context != null && requestProxy.Entries.Count > 0)
+            if (scope?.Span?.Context != null && requestProxy.Entries.Count > 0 && !string.IsNullOrEmpty(queueName))
             {
-                for (int i = 0; i < requestProxy.Entries.Count; i++)
+                var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+                var edgeTags = new[] { "direction:out", $"topic:{queueName}", "type:sqs" };
+                foreach (var e in requestProxy.Entries)
                 {
-                    var entry = requestProxy.Entries[i].DuckCast<IContainsMessageAttributes>();
-                    ContextPropagation.InjectHeadersIntoMessage<TSendMessageBatchRequest>(entry, scope.Span.Context);
+                    var entry = e.DuckCast<IContainsMessageAttributes>();
+                    // this has no effect is DSM is disabled
+                    scope.Span.SetDataStreamsCheckpoint(dataStreamsManager, CheckpointKind.Produce, edgeTags, payloadSizeBytes: 0, timeInQueueMs: 0);
```

Comment:
In `SendMessageAsyncIntegration` you check `dataStreamsManager for null: presumably we should do the same here? (if it's disabled, this is noop, but if it's null, this will throw).

Reply:
I added the check to the underlying method